shoov/modules/custom/shoov_ci_build/shoov_ci_build.module

```diff
@@ -241,7 +241,19 @@ function shoov_ci_build_message_update(Message $message) {
   // Trigger Pusher event when CI Build status changes.
   shoov_ci_build_trigger_event($message, 'ci_build_update');
 
+  $original_wrapper = entity_metadata_wrapper('message', $message->original);
+
   $status = $wrapper->field_ci_build_status->value();
+  $original_status = $original_wrapper->field_ci_build_status->value();
+  if ($status != $original_status && $status == 'in_progress') {
+    // CI Build started - save the start timestamp.
+    $wrapper->field_ci_build_start_timestamp->set(time());
+    // Update the field value on message saving.
+    field_attach_update('message', $message);
```

Comment:
this seems wrong. If you wanted to prevent recursion, you can add a custom property, and check it doesn't exist in the beginning of the function, for example:

```php
function shoov_ci_build_message_update(Message $message) {
  if (!empty($message->_skip_save)) {
    // Prevent recursion.
    return;
  }

  ...
  $wrapper->save()
}
```

However, question is why not `hook_message_presave()`?